pex/testing.py

```diff
@@ -455,6 +479,22 @@ def run_pyenv(args):
     return python, pip, run_pyenv
 
 
+def ensure_python_venv(version, latest_pip=True):
+    python, pip, _ = ensure_python_distribution(version)
+    venv = safe_mkdtemp()
+    if version in _ALL_PY3_VERSIONS:
+        subprocess.check_call([python, "-m", "venv", venv])
+    else:
+        subprocess.check_call([pip, "install", "virtualenv==16.7.10"])
+        subprocess.check_call([python, "-m", "virtualenv", venv])
+    python, pip = tuple(
+        os.path.realpath(os.path.join(venv, "bin", exe)) for exe in ("python", "pip")
+    )
+    if latest_pip:
+        subprocess.check_call([pip, "install", "-U", "pip"])
+    return python, pip
```

Comment:
Previously tests were / were encouraged to use the `pip` in a pyenv Python version. That would of course pollute that pyenv Python version with whatever was pip installed there (under `ROOT_DIR/.pyenv_test`). Now, each test method that needs to `pip` install gets a fresh venv.